controllers/chat/getSignedChannelList.js

```diff
@@ -73,10 +73,21 @@ const __filterAndTransformChannelData = async (channel) => {
   if (messageLength === 0 && channelType === CHANNEL_TYPE.TOPIC) return false;
 
   if (channelType === CHANNEL_TYPE.TOPIC) {
-    const topic = await TopicFunction.findOneByName(Topics, removePrefixTopic(newChannel.id));
-    newChannel.image = topic.icon_path || null;
-    newChannel.channelImage = topic.icon_path || null;
-    newChannel.channel_image = topic.icon_path || null;
+    const topic = await TopicFunction.getTopicDetailByTopicId(
+      sequelize,
+      userId,
+      removePrefixTopic(newChannel.id)
+    );
+
+    console.log(`topic name ${removePrefixTopic(newChannel.id)}`);
+    console.log(topic);
+
+    newChannel.image = topic?.icon_path || null;
+    newChannel.channelImage = topic?.icon_path || null;
+    newChannel.channel_image = topic?.icon_path || null;
+    newChannel.cover_image = topic?.cover_path || null;
+    newChannel.is_following = topic?.is_following || false;
+    newChannel.topic_follower_count = parseInt(topic?.topic_follower_count || 0);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes here involve fetching topic details using the `getTopicDetailByTopicId` function and updating the channel data accordingly. The use of optional chaining (`?.`) is good for handling cases where `topic` might be undefined or null. However, the console.log statements on lines 82 and 83 should be removed or replaced with proper logging mechanisms to avoid exposing potentially sensitive information in production environments.
```diff
-    const topic = await TopicFunction.findOneByName(Topics, removePrefixTopic(newChannel.id));
-    newChannel.image = topic.icon_path || null;
-    newChannel.channelImage = topic.icon_path || null;
-    newChannel.channel_image = topic.icon_path || null;
+    const topic = await TopicFunction.getTopicDetailByTopicId(
+        sequelize,
+        userId,
+        removePrefixTopic(newChannel.id)
+    );
+    // Remove these console.log statements or replace them with proper logging
+    console.log(`topic name ${removePrefixTopic(newChannel.id)}`);
+    console.log(topic);
+
+    newChannel.image = topic?.icon_path || null;
+    newChannel.channelImage = topic?.icon_path || null;
+    newChannel.channel_image = topic?.icon_path || null;
+    newChannel.cover_image = topic?.cover_path || null;
+    newChannel.is_following = topic?.is_following || false;
+    newChannel.topic_follower_count = parseInt(topic?.topic_follower_count || 0);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
will be fixed on next commit